### ISSUE TITLE
install.sh requires exemples and patch fails

### ISSUE THREAD

**Issue by robnagler** [radiasoft/zgoubi#27]
Here's the output including the contents of zgoubi/PARIZ.H.rej:

```sh
$ bash install.sh
2018-07-14 16:58:35 UTC [     info] __file: /home/vagrant/src/radiasoft/zgoubi/install.sh
2018-07-14 16:58:35 UTC [     info] __dir: /home/vagrant/src/radiasoft/zgoubi
2018-07-14 16:58:35 UTC [     info] __base: install
2018-07-14 16:58:35 UTC [     info] __os: Linux
2018-07-14 16:58:35 UTC [     info] __usage: /home/vagrant/src/radiasoft/zgoubi/install.sh-usage
2018-07-14 16:58:35 UTC [     info] LOG_LEVEL: 6
2018-07-14 16:58:35 UTC [     info] -c (--with-c):           gcc
2018-07-14 16:58:35 UTC [     info] -d (--debug):            0
2018-07-14 16:58:35 UTC [     info] -e (--verbose):          0
2018-07-14 16:58:35 UTC [     info] -f (--with-fortran):     gfortran
2018-07-14 16:58:35 UTC [     info] -h (--help):             0
2018-07-14 16:58:35 UTC [     info] -i (--install-prefix):   /home/vagrant/src/radiasoft/zgoubi/install-files/
2018-07-14 16:58:35 UTC [     info] -j (--num-threads):
2018-07-14 16:58:35 UTC [     info] -m (--with-cmake):       cmake
2018-07-14 16:58:35 UTC [     info] -n (--no-color):         0
2018-07-14 16:58:35 UTC [     info] -v (--version):          0
2018-07-14 16:58:35 UTC [     info] install_path="/home/vagrant/src/radiasoft/zgoubi/install-files"
2018-07-14 16:58:35 UTC [     info] num_threads=""
2018-07-14 16:58:35 UTC [     info] build_path="/home/vagrant/src/radiasoft/zgoubi"/build
2018-07-14 16:58:35 UTC [     info] Checking whether the directory /home/vagrant/src/radiasoft/zgoubi/install-files exists...
2018-07-14 16:58:35 UTC [     info] yes
2018-07-14 16:58:35 UTC [     info] Checking whether I have write permissions to /home/vagrant/src/radiasoft/zgoubi/install-files ...
2018-07-14 16:58:35 UTC [     info] yes
-- The C compiler identification is GNU 7.3.1
-- The Fortran compiler identification is GNU 7.3.1
-- Check for working C compiler: /usr/bin/gcc
-- Check for working C compiler: /usr/bin/gcc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working Fortran compiler: /usr/bin/gfortran
-- Check for working Fortran compiler: /usr/bin/gfortran  -- works
-- Detecting Fortran compiler ABI info
-- Detecting Fortran compiler ABI info - done
-- Checking whether /usr/bin/gfortran supports Fortran 90
-- Checking whether /usr/bin/gfortran supports Fortran 90 -- yes
patching file zgoubi/PARIZ.H
Reversed (or previously applied) patch detected!  Assume -R? [n]
Apply anyway? [n]
Skipping patch.
1 out of 1 hunk ignored -- saving rejects to file zgoubi/PARIZ.H.rej
CMake Error: File /home/vagrant/src/radiasoft/zgoubi/exemples/KEYWORDS/FIT/FIT_embedded_in_REBELOTE/centeredHelix.eps does not exist.
CMake Error at CMakeLists.txt:1138 (configure_file):
  configure_file Problem configuring file


CMake Error: File /home/vagrant/src/radiasoft/zgoubi/exemples/KEYWORDS/FIT/FIT_embedded_in_REBELOTE/centeredHelix_FIT_save.res_150226 does not exist.
CMake Error at CMakeLists.txt:1138 (configure_file):
  configure_file Problem configuring file


CMake Error: File /home/vagrant/src/radiasoft/zgoubi/exemples/KEYWORDS/FIT/FIT_embedded_in_REBELOTE/centeredHelix_FIT_save_nofinal_150226.res does not exist.
CMake Error at CMakeLists.txt:1138 (configure_file):
  configure_file Problem configuring file


CMake Error: File /home/vagrant/src/radiasoft/zgoubi/exemples/KEYWORDS/FIT/FIT_embedded_in_REBELOTE/warmSnake.map does not exist.
CMake Error at CMakeLists.txt:1138 (configure_file):
  configure_file Problem configuring file


-- Configuring incomplete, errors occurred!
See also "/home/vagrant/src/radiasoft/zgoubi/build/CMakeFiles/CMakeOutput.log".
2018-07-14 16:58:35 UTC [     info] Cleaning up. Done
$ cat zgoubi/PARIZ.H.rej
--- zgoubi/PARIZ.H
+++ zgoubi/PARIZ.H
@@ -210,7 +210,7 @@ C       PARAMETER (MXX=567, MXY=201)

 C-- Sizing of the field map array HC(ID,MXX,MXY,IZ,MMAP)
 C For iRCMS
-       PARAMETER (IZ=1 , ID=3, MMAP=1)
+       PARAMETER (IZ=29 , ID=3, MMAP=1)
        PARAMETER (MXX=2131, MXY=29)

 C---------------------------------------------------------------
```

**Comment by zbeekman** [comment]
ATTN: @rouson

**Comment by zbeekman** [comment]
On the `fix-issue-#28-fedora-27-build` branch, the `install.sh` file has been removed. Canonical CMake installation and testing should be used. The procedure for building, testing and installing with CMake and CTest inside the `beamsim-part1` docker image is documented in the script called during CI testing on Travis-CI: https://github.com/radiasoft/zgoubi/blob/e900e032aee0cbeaf547e3b66f4fefc3546f6422/travis.sh

I can run the script locally to test the build on my macOS machine (zgoubi is built inside the `beamsim-part1` docker container by executing `./travis.sh` locally, and the build can be seen at https://travis-ci.org/radiasoft/zgoubi/builds/407789840. 

For building, testing and installing outside of docker the typical, canonical CMake/CTest work flow looks something like:

```
pwd # in zgoubi root directory
export FC=/path/to/my/fortran/compiler
export CC=/path/to/my/C/compiler
mkdir build
cd build
cmake .. -DCMAKE_INSTALL_PREFIX=/path/to/where/zgoubi/will/be/installed
make -j
ctest --output-on-failure # See what went wrong when a test fails
make install # optional
```

@robnagler please review this when you have a second and let me know if this is a suitable solution.

**Comment by robnagler** [comment]
@zbeekman Thanks. Looks good. 

Does the compiler need to be specified? I thought CMake would figure that out.

**Comment by zbeekman** [comment]
It really depends on your environment and the compiler you intend to use.
If Intel compilers are present, CMake seems to prefer to use those over
GCC/GFortran. If on macOS, CMake will default to Clang over gcc (or maybe
the clang/llvm invoked with the default use of `gcc`, I don’t recall off
the top of my head.)

**Comment by robnagler** [comment]
For Fedora, we only install one compiler for each language so CMake should do the right thing. It does with our other codes.

**Comment by zbeekman** [comment]
> For Fedora, we only install one compiler for each language so CMake should do the right thing. It does with our other codes.

Yes, in [my previous comment](https://github.com/radiasoft/zgoubi/issues/27#issuecomment-407564672) I said

> For building, testing and installing outside of docker the typical, canonical CMake/CTest work flow looks something like...

The words "installing outside of docker" being critical there. If you take a look at the `travis.sh` script, it does not set `FC` or `CC` for usage with the docker image and pulls in the default system compiler which is GCC/GFortran 7.3.

**Comment by zbeekman** [comment]
@robnagler do my previous comments sufficiently clarify and resolve this issue as things stand on the `fix-issue-#28-fedora-27-build` branch? Please let me know specifically what needs to be resolved. Thanks.

**Comment by robnagler** [comment]
@zbeekman All good.
